src/mame/heathkit/h89.cpp

```diff
@@ -739,6 +771,43 @@ void h89_state::h89(machine_config & config)
 	TIMER(config, "irq_timer", 0).configure_periodic(FUNC(h89_state::h89_irq_timer), attotime::from_msec(2));
 }
 
+void h89_state::h88(machine_config &config)
+{
+	h89_base(config);
+	m_maincpu->set_io_map(&h89_state::h88_io);
+
+	HEATH_INTR_SOCKET(config, m_intr_socket, intr_ctrl_options, "original", true);
+	m_intr_socket->irq_line_cb().set_inputline(m_maincpu, INPUT_LINE_IRQ0);
+
+	// H-88-5 Cassette interface board
+	HEATH_H88_CASS(config, m_cassette, H89_CLOCK);
+}
+
+void h89_state::h89(machine_config &config)
+{
+	h89_base(config);
+	m_maincpu->set_io_map(&h89_state::h89_io);
+
+	HEATH_INTR_SOCKET(config, m_intr_socket, intr_ctrl_options, "h37", true);
+	m_intr_socket->irq_line_cb().set_inputline(m_maincpu, INPUT_LINE_IRQ0);
+
+	// Z-89-37 Soft-sectored controller
+	HEATH_Z37_FDC(config, m_h37);
+	m_h37->drq_cb().set(m_intr_socket, FUNC(heath_intr_socket::set_drq));
+	m_h37->irq_cb().set(m_intr_socket, FUNC(heath_intr_socket::set_irq));
+	m_h37->block_interrupt_cb().set(m_intr_socket, FUNC(heath_intr_socket::block_interrupts));
+}
```

Comment:
You could add the interrupt controller in the base function and just use `set_default_option` to set the default interrupt controller for each concrete machine.